File: JavaScript/Project/MVVMtpl_WinJS/MVVMtpl_WinJS.Windows/bin/Debug/AppX/Services/NetworkService.js
(function (g, WinJS) {
    "use strict";

    var networkInfo = Windows.Networking.Connectivity.NetworkInformation; 
    var networkConnectivityInfo = Windows.Networking.Connectivity.NetworkConnectivityLevel; 

    var NetworkService = WinJS.Class.define(
        function () {
        },

        {
            isOnline: function() { 
                var connectionProfile = networkInfo.getInternetConnectionProfile(); 
                if (connectionProfile === null) { 
                    return false; 
                }

                var networkConnectivityLevel = connectionProfile.getNetworkConnectivityLevel(); 
                if (networkConnectivityLevel === networkConnectivityInfo.none 
                || networkConnectivityLevel === networkConnectivityInfo.localAccess 
                || networkConnectivityLevel === networkConnectivityInfo.constrainedInternetAccess) { 

                    return false;
                } 

                return true; 
            }, 

        }
    );

    WinJS.Namespace.define("Services", {
        NetworkService: NetworkService
    });
})(this, WinJS);

File: CSharp/Project/MVVMtpl/MVVMtpl.Shared/Services/NetworkService.cs


namespace MVVMtpl.Services
{
    using System;
    using System.Collections.Generic;
    using System.Net.NetworkInformation;
    using System.Text;
    using Windows.Networking.Connectivity;

    public class NetworkService
    {
        public bool IsOnline()
        {
            var internetProfile = NetworkInformation.GetInternetConnectionProfile();
            return internetProfile != null &&
                internetProfile.GetNetworkConnectivityLevel() == NetworkConnectivityLevel.InternetAccess;            
        }

    }
}
